### ISSUE TITLE
Hbase compatibility error in pinpoint-hbase2-batch

### ISSUE THREAD

**Issue by dv-morais** [pinpoint-apm/pinpoint#8901]
Version: 2.4.0
Hbase: 2.3.0

I'm trying to execute the pinpoint-hbase2-batch but I'm receiving compatibility errors saying that only versions 0.9 or 1 are compatible...

See log below.

```
05-31 14:39:17.203 INFO  ProfileResolveListener              : onApplicationEvent(ApplicationEnvironmentPreparedEvent)
05-31 14:39:17.205 INFO  ProfileEnvironmentPostProcessor     : postProcessEnvironment
05-31 14:39:17.207 INFO  PinpointProfileEnvironment          : spring.profiles.active:[release]
05-31 14:39:17.207 INFO  PinpointProfileEnvironment          : pinpoint.profiles.active=release
05-31 14:39:17.208 INFO  PinpointProfileEnvironment          : PropertiesPropertySource pinpoint.profiles.active=release
05-31 14:39:17.208 INFO  PinpointProfileEnvironment          : PropertiesPropertySource logging.config=classpath:profiles/release/log4j2.xml
05-31 14:39:17.209 INFO  PinpointProfileEnvironment          : Add PropertySource name:PinpointProfileEnvironment 'pinpoint.profiles.active=release'
2022-05-31 14:39:17,333 main INFO The nolookups option will be ignored. Message Lookups are no longer supported.
2022-05-31 14:39:17,349 main INFO The nolookups option will be ignored. Message Lookups are no longer supported.
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- applicationEvent:ApplicationEnvironmentPreparedEvent
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name configurationProperties
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name servletConfigInitParams
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name servletContextInitParams
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name systemProperties
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name systemEnvironment
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name random
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name Config resource 'file [config/application.properties]' via location 'optional:file:./config/'
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name Config resource 'class path resource [application.yml]' via location 'optional:classpath:/'
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name PinpointProfileEnvironment 'pinpoint.profiles.active=release'
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.ExternalEnvironmentListener  -- onApplicationEvent(ApplicationEnvironmentPreparedEvent})
05-31 14:39:17.017 [           main] INFO  c.n.p.c.s.e.ExternalEnvironment          -- -Dpinpoint.batch.config.location is not set
05-31 14:39:17.017 [           main] INFO  c.n.p.b.BatchApp                         -- Starting BatchApp v2.4.0 using Java 11.0.13 on apmweb with PID 4024535 (/web/java/pinpoint-web/pinpoint-hbase2-batch-2.4.0.jar started by root in /web/java/pinpoint-web)
05-31 14:39:17.017 [           main] INFO  c.n.p.b.BatchApp                         -- The following 1 profile is active: "release"
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'hbaseConfiguration' of type [com.navercorp.pinpoint.common.hbase.HbaseConfigurationFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'hbaseConfiguration' of type [org.apache.hadoop.conf.Configuration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'hbaseThreadPool' of type [com.navercorp.pinpoint.common.server.util.PinpointThreadPoolExecutorFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'hbaseThreadPool' of type [java.util.concurrent.ThreadPoolExecutor] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  c.n.p.b.c.BatchConfiguration             -- before setup field: BatchConfiguration{emailServerUrl='smtpprdm.prodemge.gov.br', senderEmailAddress='pinpoint@prodemge.gov.br', enableWebhook='true', pinpointUrl='http://apm.prodemge.gov.br', batchEnv='release', flinkServerList=[10.100.40.109], enableCleanupInactiveAgents=false, cleanupInactiveAgentsDurationDays=30, cleanupInactiveAgentsCron=''}
05-31 14:39:19.019 [           main] INFO  c.n.p.b.c.BatchConfiguration             -- after setup field : BatchConfiguration{emailServerUrl='smtpprdm.prodemge.gov.br', senderEmailAddress='pinpoint@prodemge.gov.br', enableWebhook='true', pinpointUrl='http://apm.prodemge.gov.br', batchEnv='release', flinkServerList=[10.100.40.109], enableCleanupInactiveAgents=false, cleanupInactiveAgentsDurationDays=30, cleanupInactiveAgentsCron='0 0 0 29 2 ?'}
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'batchConfiguration' of type [com.navercorp.pinpoint.batch.common.BatchConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'com.fasterxml.jackson.databind.ObjectMapper#7966baa7' of type [com.fasterxml.jackson.databind.ObjectMapper] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'org.springframework.http.converter.json.MappingJackson2HttpMessageConverter#1253e7cb' of type [org.springframework.http.converter.json.MappingJackson2HttpMessageConverter] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'org.springframework.http.converter.FormHttpMessageConverter#2d36e77e' of type [org.springframework.http.converter.FormHttpMessageConverter] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'org.springframework.http.converter.StringHttpMessageConverter#61c9c3fd' of type [org.springframework.http.converter.StringHttpMessageConverter] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'restTemplate' of type [org.springframework.web.client.RestTemplate] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'driverProperties' of type [org.springframework.beans.factory.config.PropertiesFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'driverProperties' of type [java.util.Properties] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
Loading class `com.mysql.jdbc.Driver'. This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'. The driver is automatically registered via the SPI and manual loading of the driver class is generally unnecessary.
05-31 14:39:19.019 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'dataSourceHikariConfig' of type [com.zaxxer.hikari.HikariConfig] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:19.019 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-dataSource - Starting...
05-31 14:39:19.019 [           main] WARN  c.z.h.u.DriverDataSource                 -- Registered driver with driverClassName=com.mysql.jdbc.Driver was not found, trying direct instantiation.
05-31 14:39:20.020 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-dataSource - Start completed.
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'dataSource' of type [com.zaxxer.hikari.HikariDataSource] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'com.navercorp.pinpoint.web.dao.ibatis.BindingLogPlugin32#4c5474f5' of type [com.navercorp.pinpoint.web.dao.ibatis.BindingLogPlugin32] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'sqlSessionFactory' of type [org.mybatis.spring.SqlSessionFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'sqlSessionFactory' of type [org.apache.ibatis.session.defaults.DefaultSqlSessionFactory] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'sqlSessionTemplate' of type [org.mybatis.spring.SqlSessionTemplate] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'userDao' of type [com.navercorp.pinpoint.web.dao.mysql.MysqlUserDao] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'userServiceImpl' of type [com.navercorp.pinpoint.web.service.UserServiceImpl] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'webhookDao' of type [com.navercorp.pinpoint.web.dao.mysql.MysqlWebhookDao] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'webhookSendInfoDao' of type [com.navercorp.pinpoint.web.dao.mysql.MysqlWebhookSendInfoDao] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'webhookServiceImpl' of type [com.navercorp.pinpoint.web.service.WebhookServiceImpl] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  trationDelegate$BeanPostProcessorChecker -- Bean 'jobRegistry' of type [org.springframework.batch.core.configuration.support.MapJobRegistry] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
05-31 14:39:20.020 [           main] INFO  o.s.b.w.e.t.TomcatWebServer              -- Tomcat initialized with port(s): 8090 (http)
05-31 14:39:20.020 [           main] INFO  o.a.c.h.Http11NioProtocol                -- Initializing ProtocolHandler ["http-nio-8090"]
05-31 14:39:20.020 [           main] INFO  o.a.c.c.StandardService                  -- Starting service [Tomcat]
05-31 14:39:20.020 [           main] INFO  o.a.c.c.StandardEngine                   -- Starting Servlet engine: [Apache Tomcat/9.0.60]
05-31 14:39:20.020 [           main] INFO  o.a.c.c.C.[.[.[/]                        -- Initializing Spring embedded WebApplicationContext
05-31 14:39:20.020 [           main] INFO  w.s.c.ServletWebServerApplicationContext -- Root WebApplicationContext: initialization completed in 3306 ms
05-31 14:39:20.020 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-meta - Starting...
05-31 14:39:20.020 [           main] WARN  c.z.h.u.DriverDataSource                 -- Registered driver with driverClassName=com.mysql.jdbc.Driver was not found, trying direct instantiation.
05-31 14:39:20.020 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-meta - Start completed.
05-31 14:39:20.020 [           main] INFO  o.s.b.c.r.s.JobRepositoryFactoryBean     -- No database type set, using meta data indicating: MYSQL
05-31 14:39:21.021 [           main] INFO  o.s.b.c.l.s.SimpleJobLauncher            -- No TaskExecutor has been set, defaulting to synchronous executor.
05-31 14:39:21.021 [           main] INFO  c.n.p.w.c.ConfigProperties               -- ConfigProperties{sendUsage=true, editUserInfo=true, showActiveThread=false, showActiveThreadDump=false, enableActiveThreadDump=false, enableServerMapRealTime=false, openSource=true, securityGuideUrl='null', showApplicationStat=true, showStackTraceOnError=true, webSocketAllowedOrigins=*, webhookEnable=true}
05-31 14:39:21.021 [           main] WARN  o.a.h.u.NativeCodeLoader                 -- Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.hadoop.security.authentication.util.KerberosUtil (jar:file:/web/java/pinpoint-web/pinpoint-hbase2-batch-2.4.0.jar!/BOOT-INF/lib/hbase-shaded-client-1.7.1.jar!/) to method sun.security.krb5.Config.getInstance()
WARNING: Please consider reporting this to the maintainers of org.apache.hadoop.security.authentication.util.KerberosUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
05-31 14:39:21.021 [           main] INFO  o.a.h.h.z.RecoverableZooKeeper           -- Process identifier=hconnection-0x64b7225f connecting to ZooKeeper ensemble=apmbanco:2181
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:zookeeper.version=3.4.10-39d3a4f269333c922ed3db283be479f9deacaa0f, built on 03/23/2017 10:13 GMT
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:host.name=apmweb
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.version=11.0.13
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.vendor=Amazon.com Inc.
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.home=/usr/lib/jvm/java-11-amazon-corretto
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.class.path=pinpoint-hbase2-batch-2.4.0.jar
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.library.path=/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.io.tmpdir=/tmp
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:java.compiler=<NA>
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:os.name=Linux
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:os.arch=amd64
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:os.version=4.18.0-193.el8.x86_64
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:user.name=root
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:user.home=/root
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Client environment:user.dir=/web/java/pinpoint-web
05-31 14:39:21.021 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Initiating client connection, connectString=apmbanco:2181 sessionTimeout=90000 watcher=org.apache.hadoop.hbase.zookeeper.PendingWatcher@66153688
05-31 14:39:21.021 [(apmbanco:2181)] INFO  o.a.h.h.s.o.a.z.ClientCnxn               -- Opening socket connection to server apmbanco/10.100.40.113:2181. Will not attempt to authenticate using SASL (unknown error)
05-31 14:39:21.021 [(apmbanco:2181)] INFO  o.a.h.h.s.o.a.z.ClientCnxn               -- Socket connection established to apmbanco/10.100.40.113:2181, initiating session
05-31 14:39:21.021 [(apmbanco:2181)] INFO  o.a.h.h.s.o.a.z.ClientCnxn               -- Session establishment complete on server apmbanco/10.100.40.113:2181, sessionid = 0x1001b229d8b0013, negotiated timeout = 90000
log4j:WARN No appenders could be found for logger (org.apache.htrace.core.Tracer).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
05-31 14:39:22.022 [           main] INFO  c.n.p.c.h.HbaseVersionCheckBean          -- hbase-clusterStatus:Master: apmbanco,16000,1653414280403
Number of backup masters: 0
Number of live region servers: 1
  apmbanco,16020,1653414281411
Number of dead region servers: 0
Average load: 486.0
Number of requests: 1577
Number of regions: 486
Number of regions in transition: 0
05-31 14:39:22.022 [           main] INFO  c.n.p.c.h.HbaseVersionCheckBean          -- hbase-clusterStatus HBaseServerVersion:2.3.0
05-31 14:39:22.022 [           main] INFO  c.n.p.c.h.HbaseVersionCheckBean          -- HBaseClientVersion:1.7.1
05-31 14:39:22.022 [           main] ERROR c.n.p.c.h.HbaseVersionCheckBean          -- HBase version compatibility violation HBaseClient:HBaseVersion{supportedVersion=[0.9, 1.]} V1, HBaseServer:2.3.0
05-31 14:39:22.022 [           main] WARN  ConfigServletWebServerApplicationContext -- Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.navercorp.pinpoint.common.hbase.HbaseVersionCheckBean#0' defined in class path resource [applicationContext-web-hbase.xml]: Invocation of init method failed; nested exception is com.navercorp.pinpoint.common.hbase.HBaseAccessException: HBase version compatibility violation HBaseClient:HBaseVersion{supportedVersion=[0.9, 1.]} V1, HBaseServer:2.3.0
05-31 14:39:22.022 [           main] INFO  c.n.p.c.h.HbaseTemplate2                 -- HBaseTemplate2.destroy()
05-31 14:39:22.022 [           main] INFO  c.n.p.c.h.ConnectionFactoryBean          -- Hbase Connection destroy()
05-31 14:39:22.022 [           main] INFO  nectionManager$HConnectionImplementation -- Closing master protocol: MasterService
05-31 14:39:22.022 [           main] INFO  nectionManager$HConnectionImplementation -- Closing zookeeper sessionid=0x1001b229d8b0013
05-31 14:39:22.022 [           main] INFO  o.a.h.h.s.o.a.z.ZooKeeper                -- Session: 0x1001b229d8b0013 closed
05-31 14:39:22.022 [ain-EventThread] INFO  o.a.h.h.s.o.a.z.ClientCnxn               -- EventThread shut down for session: 0x1001b229d8b0013
05-31 14:39:22.022 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-meta - Shutdown initiated...
05-31 14:39:22.022 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-meta - Shutdown completed.
05-31 14:39:22.022 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-dataSource - Shutdown initiated...
05-31 14:39:22.022 [           main] INFO  c.z.h.HikariDataSource                   -- HikariPool-dataSource - Shutdown completed.
05-31 14:39:22.022 [           main] INFO  o.a.c.c.StandardService                  -- Stopping service [Tomcat]
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- applicationEvent:ApplicationFailedEvent
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name configurationProperties
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name servletConfigInitParams
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name servletContextInitParams
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name systemProperties
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name systemEnvironment
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name random
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name Config resource 'file [config/application.properties]' via location 'optional:file:./config/'
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name Config resource 'class path resource [application.yml]' via location 'optional:classpath:/'
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- PropertySource name PinpointProfileEnvironment 'pinpoint.profiles.active=release'
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- CompositePropertySource name BatchAppPropertySources-JDBC
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   --   class path resource [profiles/release/jdbc.properties]
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   --   class path resource [jdbc-root.properties]
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- CompositePropertySource name BatchAppPropertySources-HBase
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   --   class path resource [profiles/release/hbase.properties]
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   --   class path resource [hbase-root.properties]
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   -- CompositePropertySource name BatchAppPropertySources
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   --   class path resource [profiles/release/batch.properties]
05-31 14:39:22.022 [           main] INFO  c.n.p.c.s.e.EnvironmentLoggingListener   --   class path resource [batch-root.properties]
05-31 14:39:22.022 [           main] INFO  ConditionEvaluationReportLoggingListener -- 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
05-31 14:39:22.022 [           main] ERROR o.s.b.SpringApplication                  -- Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.navercorp.pinpoint.common.hbase.HbaseVersionCheckBean#0' defined in class path resource [applicationContext-web-hbase.xml]: Invocation of init method failed; nested exception is com.navercorp.pinpoint.common.hbase.HBaseAccessException: HBase version compatibility violation HBaseClient:HBaseVersion{supportedVersion=[0.9, 1.]} V1, HBaseServer:2.3.0
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918) ~[spring-context-5.3.20.jar!/:5.3.20]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.20.jar!/:5.3.20]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145) ~[spring-boot-2.5.12.jar!/:2.5.12]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:775) [spring-boot-2.5.12.jar!/:2.5.12]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:448) [spring-boot-2.5.12.jar!/:2.5.12]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:339) [spring-boot-2.5.12.jar!/:2.5.12]
	at com.navercorp.pinpoint.common.server.starter.BasicStarter.start(BasicStarter.java:53) [pinpoint-commons-server-2.4.0.jar!/:2.4.0]
	at com.navercorp.pinpoint.batch.BatchApp.main(BatchApp.java:46) [pinpoint-batch-2.4.0.jar!/:2.4.0]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) [pinpoint-hbase2-batch-2.4.0.jar:2.4.0]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) [pinpoint-hbase2-batch-2.4.0.jar:2.4.0]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) [pinpoint-hbase2-batch-2.4.0.jar:2.4.0]
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) [pinpoint-hbase2-batch-2.4.0.jar:2.4.0]
Caused by: com.navercorp.pinpoint.common.hbase.HBaseAccessException: HBase version compatibility violation HBaseClient:HBaseVersion{supportedVersion=[0.9, 1.]} V1, HBaseServer:2.3.0
	at com.navercorp.pinpoint.common.hbase.HbaseVersionCheckBean.afterPropertiesSet(HbaseVersionCheckBean.java:57) ~[pinpoint-commons-hbase-2.4.0.jar!/:2.4.0]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863) ~[spring-beans-5.3.20.jar!/:5.3.20]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800) ~[spring-beans-5.3.20.jar!/:5.3.20]
	... 23 more
05-31 14:39:22.209 ERROR BatchApp                            : [WebApp] could not launch app.
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.navercorp.pinpoint.common.hbase.HbaseVersionCheckBean#0' defined in class path resource [applicationContext-web-hbase.xml]: Invocation of init method failed; nested exception is com.navercorp.pinpoint.common.hbase.HBaseAccessException: HBase version compatibility violation HBaseClient:HBaseVersion{supportedVersion=[0.9, 1.]} V1, HBaseServer:2.3.0
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:775)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:448)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:339)
	at com.navercorp.pinpoint.common.server.starter.BasicStarter.start(BasicStarter.java:53)
	at com.navercorp.pinpoint.batch.BatchApp.main(BatchApp.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
Caused by: com.navercorp.pinpoint.common.hbase.HBaseAccessException: HBase version compatibility violation HBaseClient:HBaseVersion{supportedVersion=[0.9, 1.]} V1, HBaseServer:2.3.0
	at com.navercorp.pinpoint.common.hbase.HbaseVersionCheckBean.afterPropertiesSet(HbaseVersionCheckBean.java:57)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 23 more
```

**Comment by emeroad** [comment]
This is a bug in the hbase2-batch module. 
The hbase-client 2.x has not been updated on the batch.
This bug will be fixed in 2.4.1.

**Comment by minwoo-jung** [comment]
https://github.com/pinpoint-apm/pinpoint/pull/8920
